diff --git a/load.c b/load.c
index 3a6c7ef..0d25cbc 100644
--- a/load.c
+++ b/load.c
@@ -753,7 +753,6 @@ static int FindStart(int firsttime)
 void LoadSGF(struct SGFInfo *sgf)
 {
 	long size;
-	int miss;
 
 	sgfc = sgf;			/* set current SGFInfo context */
 
@@ -781,17 +780,26 @@ void LoadSGF(struct SGFInfo *sgf)
 	sgf->b_end   = sgf->buffer + size;
 	sgf->current = sgf->buffer;
 
-	miss = FindStart(TRUE);		/* skip junk in front of '(;' */
-	sgf->start = sgf->current;
+  LoadSGFFromFileBuffer(sgf);
+}
 
-	while(!SGF_EOF)
-	{
-		if(!miss)
-			sgf->current++;			/* skip '(' */
-		if(!BuildSGFTree(NULL))
-			break;
-		miss = FindStart(FALSE);	/* skip junk in front of '(;' */
-	}
+void LoadSGFFromFileBuffer(struct SGFInfo *sgf)
+{
+  int miss;
+
+  sgfc = sgf;      /* set current SGFInfo context */
+
+  miss = FindStart(TRUE);    /* skip junk in front of '(;' */
+  sgf->start = sgf->current;
+
+  while(!SGF_EOF)
+  {
+    if(!miss)
+      sgf->current++;      /* skip '(' */
+    if(!BuildSGFTree(NULL))
+      break;
+    miss = FindStart(FALSE);  /* skip junk in front of '(;' */
+  }
 
-	PrintError(E_NO_ERROR);			/* flush accumulated messages */
+  PrintError(E_NO_ERROR);      /* flush accumulated messages */
 }
diff --git a/protos.h b/protos.h
index 5ea0d40..4accc01 100644
--- a/protos.h
+++ b/protos.h
@@ -50,6 +50,8 @@ struct Node *NewNode(struct Node * , int);
 char *SkipText(char * , char * , char , int );
 void LoadSGF(struct SGFInfo * );
 
+void LoadSGFFromFileBuffer(struct SGFInfo * );
+
 
 /**** save.c ****/
 
